### PR TITLE
Clean up the `Runtime` base class

### PR DESCRIPTION
**Pull request by gctucker**
Remove unused code from the base `kernelci.runtime.Runtime` now that we have `shell`, `docker` and `kubernetes` runtimes implemented.

### CODE REVIEW

File: kernelci/runtime/kubernetes.py [hunk 1]
```diff
@@ -22,18 +22,19 @@ class Kubernetes(Runtime):
 
     JOB_NAME_CHARACTERS = string.ascii_lowercase + string.digits
 
-    def generate(self, params, plan_config):
-        template = self._get_template(plan_config)
+    @classmethod
+    def _get_job_file_name(cls, params):
+        return '.'.join([params['k8s_job_name'], 'yaml'])
```
Comment (mgalka) [inline kernelci/runtime/kubernetes.py:27]: isn't it the same as `params['k8s_job_name'] + '.yaml'`?

File: kernelci/runtime/kubernetes.py [hunk 2]
```diff
@@ -22,18 +22,19 @@ class Kubernetes(Runtime):
...
+
+    def generate(self, params, job_config):
+        template = self._get_template(job_config)
         job_name = '-'.join(['kci', params['node_id'], params['name'][:24]])
         safe_name = re.sub(r'[\:/_+=]', '-', job_name).lower()
         rand_sx = ''.join(random.sample(self.JOB_NAME_CHARACTERS, 8))
         k8s_job_name = '-'.join([safe_name[:(62 - len(rand_sx))], rand_sx])
```
Comment (mgalka) [inline kernelci/runtime/kubernetes.py:34]: I'd consider extracting it to a separate function.

File: kernelci/runtime/kubernetes.py (hunk 1 above)
Comment (gctucker) [reply kernelci/runtime/kubernetes.py:27]: I've always thought that join was better and faster than + for joining strings so I'm always using this approach. I think there's an article somewhere comparing these things but I read it several years ago.

File: kernelci/runtime/kubernetes.py (hunk 2 above)
Comment (gctucker) [reply kernelci/runtime/kubernetes.py:34]: That's unrelated to this PR. Feel free to send a change to rework this if you want.

File: kernelci/runtime/kubernetes.py (hunk 1 above)
Comment (gctucker) [reply kernelci/runtime/kubernetes.py:27]: Also it's not a change related to this PR either, it's only reworking the method as a private class one.

Comment (mgalka) [reply kernelci/runtime/kubernetes.py:27]: > I've always thought that join was better and faster than + for joining strings so I'm always using this approach. I think there's an article somewhere comparing these things but I read it several years ago.

Yes, in general it is, I think still the fastest method of string concatenation and + is the slowest. I put '+' expression there as an example of what I have in mind. Maybe not clearly enough. `.join` is the fastest, slower, but still fast is `f-string`, the slowest is `+`. However, in this case we're joining 2 strings and the speed is not the priority, but "readability counts` principle is still in use. This is just for the sake of clarification.

If it's not part of your PR, feel free to ignore it. I'll possible convert it to an f-string at some other occasion.

File: kernelci/runtime/kubernetes.py (hunk 2 above)
Comment (mgalka) [reply kernelci/runtime/kubernetes.py:34]: ack

File: kernelci/runtime/kubernetes.py (hunk 1 above)
Comment (gctucker) [reply kernelci/runtime/kubernetes.py:27]: Right, well once we have the whole pipeline actually functional then we can review the whole code and tweak tiny details like this I guess.

**Pull request merged**